woocommerce/woocommerce-ios
# Update quantity when scanning products already in order

Closes: #9784

## Description
Before this PR, scanning the same barcode multiple times would add multiple product rows to the order. With this change we will update the product's quantity, rather than adding an additional row, if a product already exists in the current Order.

## Changes

Rather than sending the product directly to the remote sync, we perform a check to see if this already exists within the order, if that's the case, we use the pre-existing `incrementQuantity()` method to increase it. Otherwise, we'll send it to the order as before.

## Testing instructions
On a physical device: 
- Go to `Order` > `+` > Tap the scan button > scan a barcode for a product (not a variation) > See how is added to the current order
- Tap the scan button again > scan the same barcode > See how the quantity is increased

## Screenshots

<img width="450" alt="Screenshot 2023-05-24 at 09 38 04" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/66469148-3b68-4dd9-949e-b675e2dd23cf">
